### PR TITLE
Get channels no re

### PR DESCRIPTION
**Pull request by will-moore**
Resolves a long-standing issue of unnecessarily loading the Rendering Engine when displaying the right "General" panel.

Blitz Gateway image.getChannels() loads the RE by default, allowing channel.getColor() etc. but this is not needed for the General panel.
A new method, image.getChannelsNoRE() does not attempt to load the rendering engine and this is now used for the General pane.

To test:
 - Import a big image (jpeg/png) and try loading General panel during Pyramid generation (when rendering engine throws MissingPyramidException. Panel should load without exception and show channel names.
 - Test that editing channel names works.
 - Also try viewing channel names for SPW images.

### CODE REVIEW

**Comment by will-moore** [comment]
cc @chris-allan Blitz API addition etc.

File: components/tools/OmeroPy/src/omero/gateway/__init__.py [hunk 1]
```diff
@@ -7454,15 +7458,26 @@ def getChannels(self):
...
-            params.map = {"pid": rlong(pid)}
-            query = ("select p from Pixels p join fetch p.channels as c "
-                     "join fetch c.logicalChannel as lc where p.id=:pid")
-            pixels = self._conn.getQueryService().findByQuery(
-                query, params, self._conn.SERVICE_OPTS)
-            return [ChannelWrapper(self._conn, c, idx=n, re=self._re, img=self)
-                    for n, c in enumerate(pixels.iterateChannels())]
+            return self.getChannelsNoRE()
+
+    def getChannelsNoRE(self):
```
Comment (joshmoore) [inline components/tools/OmeroPy/src/omero/gateway/__init__.py]: Is there a reason to have a new method here but a new parameter on `channelMetadata`? My gut feeling would be to do the same in both places.

Comment (will-moore) [inline components/tools/OmeroPy/src/omero/gateway/__init__.py]: Agreed that would be ideal.
I guess I was a little reluctant to mess with the class-based ```@assert_re``` decorator on image.getChannels() that would have to be removed and the same functionality applied within the method itself only if NoRE is True.
But I'll see if this is possible without breaking anything.

**Comment by joshmoore** [comment]
Any ideas on this, @will-moore ?

**Comment by will-moore** [comment]
@joshmoore Added ```noRE``` to ```image.getChannels(noRE=False)``` and added a test.

**Comment by sbesson** [comment]
I may be missing something about `test_missing_pyramid.py` as I do not see any element of logic to generate an image with a missing pyramid. The `setUp` fixture is using the gateway pytest `author_testimg_generated` which is creating a regular test image.

**Comment by will-moore** [comment]
@sbesson This line: ``` self.image._conn.createRenderingEngine = lambda: None ``` means that the image will not have a rendering engine created when image._prepareRE() is called.

**Comment by sbesson** [comment]
Got it. Would it be worth renaming this file or exanding the docstrings? The names `test_missing_pyramid/TestPyramid` convey the wrong message as the tests are more about testing/mocking the BlitzGateway rendering engine.

**Comment by will-moore** [comment]
I didn't create or name these tests originally, so I'd defer to @chris-allan or @joshmoore.
But these tests are under /gatewaytest/ directory so they are mostly testing Blitz Gateway logic.

**Comment by joshmoore** [comment]
```
./components/tools/OmeroPy/src/omero/gateway/__init__.py:7464:80: E501 line too long (83 > 79 characters)
```

**Comment by sbesson** [comment]
This PR is likely related to the following failure https://ci.openmicroscopy.org/view/Failing/job/OMERO-5.1-merge-integration-python/428/testReport/test.integration.gatewaytest.test_image/TestImage/testProperties/

**Comment by pwalczysko** [comment]
Bug: 
  - Import 8k x 8k.jpg image. 
  - Wait till pyramids are generated and the thumb is present. 
  - In General pane, change the name of first channel and 
  - hit ``Apply to All``
  - observe error

```
Traceback (most recent call last):

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/django/core/handlers/base.py", line 114, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/django/views/decorators/http.py", line 41, in inner
    return func(request, *args, **kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/decorators.py", line 471, in wrapped
    retval = f(request, *args, **kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/decorators.py", line 521, in wrapper
    context = f(request, *args, **kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/webclient/views.py", line 2112, in edit_channel_names
    ptype, [pid], nameDict, channelCount=sizeC)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omero/gateway/__init__.py", line 3506, in setChannelNames
    self.getUpdateService().saveCollection(toSave, ctx)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omero/gateway/__init__.py", line 4124, in __call__
    return self.handle_exception(e, *args, **kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omeroweb/webclient/webclient_gateway.py", line 2066, in handle_exception
    e, *args, **kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omero/gateway/__init__.py", line 4121, in __call__
    return self.f(*args, **kwargs)

  File "/opt/hudson/workspace/OMERO-5.1-merge-deploy/src/dist/lib/python/omero_api_IUpdate_ice.py", line 119, in saveCollection
    return _M_omero.api.IUpdate._op_saveCollection.invoke(self, ((objs, ), _ctx))

OptimisticLockException: exception ::omero::OptimisticLockException
{
    serverStackTrace = ome.conditions.OptimisticLockException: You are not authorized to change the update event for ome.model.core.LogicalChannel:Id_26747 from ome.model.meta.Event:Id_436479 to ome.model.meta.Event:Id_435864
You may need to reload the object before continuing.
	at ome.security.basic.OmeroInterceptor.managedEvent(OmeroInterceptor.java:1080)
	at ome.security.basic.OmeroInterceptor.checkManagedDetails(OmeroInterceptor.java:850)
	at ome.security.basic.OmeroInterceptor.resetDetails(OmeroInterceptor.java:393)
	at ome.security.basic.OmeroInterceptor.onFlushDirty(OmeroInterceptor.java:188)
	at org.hibernate.event.def.DefaultFlushEntityEventListener.invokeInterceptor(DefaultFlushEntityEventListener.java:372)
	at org.hibernate.event.def.DefaultFlushEntityEventListener.handleInterception(DefaultFlushEntityEventListener.java:349)
	at org.hibernate.event.def.DefaultFlushEntityEventListener.scheduleUpdate(DefaultFlushEntityEventListener.java:287)
	at org.hibernate.event.def.DefaultFlushEntityEventListener.onFlushEntity(DefaultFlushEntityEventListener.java:155)
	at org.hibernate.event.def.AbstractFlushingEventListener.flushEntities(AbstractFlushingEventListener.java:219)
	at org.hibernate.event.def.AbstractFlushingEventListener.flushEverythingToExecutions(AbstractFlushingEventListener.java:99)
	at org.hibernate.event.def.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:50)
	at org.hibernate.impl.SessionImpl.flush(SessionImpl.java:1208)
	at ome.logic.UpdateImpl.afterUpdate(UpdateImpl.java:295)
	at ome.logic.UpdateImpl.doAction(UpdateImpl.java:313)
	at ome.logic.UpdateImpl.doAction(UpdateImpl.java:303)
	at ome.logic.UpdateImpl.saveCollection(UpdateImpl.java:128)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
	at ome.security.basic.EventHandler.invoke(EventHandler.java:154)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.orm.hibernate3.HibernateInterceptor.invoke(HibernateInterceptor.java:111)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:108)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at ome.tools.hibernate.ProxyCleanupFilter$Interceptor.invoke(ProxyCleanupFilter.java:249)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at ome.services.util.ServiceHandler.invoke(ServiceHandler.java:121)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy93.saveCollection(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
	at ome.security.basic.BasicSecurityWiring.invoke(BasicSecurityWiring.java:98)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at ome.services.blitz.fire.AopContextInitializer.invoke(AopContextInitializer.java:43)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy93.saveCollection(Unknown Source)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at ome.services.blitz.util.IceMethodInvoker.invoke(IceMethodInvoker.java:179)
	at ome.services.throttling.Callback.run(Callback.java:56)
	at ome.services.throttling.InThreadThrottlingStrategy.callInvokerOnRawArgs(InThreadThrottlingStrategy.java:56)
	at ome.services.blitz.impl.AbstractAmdServant.callInvokerOnRawArgs(AbstractAmdServant.java:149)
	at ome.services.blitz.impl.UpdateI.saveCollection_async(UpdateI.java:76)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:307)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
	at omero.cmd.CallContext.invoke(CallContext.java:78)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:202)
	at com.sun.proxy.$Proxy94.saveCollection_async(Unknown Source)
	at omero.api._IUpdateTie.saveCollection_async(_IUpdateTie.java:117)
	at omero.api._IUpdateDisp.___saveCollection(_IUpdateDisp.java:179)
	at omero.api._IUpdateDisp.__dispatch(_IUpdateDisp.java:391)
	at IceInternal.Incoming.invoke(Incoming.java:159)
	at Ice.ConnectionI.invokeAll(ConnectionI.java:2357)
	at Ice.ConnectionI.dispatch(ConnectionI.java:1208)
	at Ice.ConnectionI.message(ConnectionI.java:1163)
	at IceInternal.ThreadPool.run(ThreadPool.java:302)
	at IceInternal.ThreadPool.access$300(ThreadPool.java:12)
	at IceInternal.ThreadPool$EventHandlerThread.run(ThreadPool.java:643)
	at java.lang.Thread.run(Thread.java:744)

    serverExceptionClass = ome.conditions.OptimisticLockException
    message = You are not authorized to change the update event for ome.model.core.LogicalChannel:Id_26747 from ome.model.meta.Event:Id_436479 to ome.model.meta.Event:Id_435864
You may need to reload the object before continuing.
    backOff = 0
}


<WSGIRequest
path:/merge/webclient/edit_channel_names/253141/,
GET:<QueryDict: {}>,
POST:<QueryDict: {u'channel2': [u'Blue'], u'channel1': [u'Green'], u'channel0': [u'test change'], u'confirm_apply': [u'apply'], u'parentId': [u'dataset-23861']}>,
COOKIES:{'_ga': 'GA1.2.483423307.1416590819',
 'csrftoken': 'CFDjgGaVwzFkmVwBxRDBt0OMRe1ClEx4',
 'phpbb3_a9mfv_k': '31e5b68a4d3f5896',
 'phpbb3_a9mfv_sid': 'e804e7bdcfddebcbb684fb6e98fccb1a',
 'phpbb3_a9mfv_u': '2759',
 'sessionid_merge': '4borjmxvx1kpwfnpveua37thh3zi211r',
 'style_cookie': 'null'},
META:{'CONTENT_LENGTH': '92',
 'CONTENT_TYPE': 'application/x-www-form-urlencoded; charset=UTF-8',
 u'CSRF_COOKIE': u'CFDjgGaVwzFkmVwBxRDBt0OMRe1ClEx4',
 'HTTPS': 'on',
 'HTTP_ACCEPT': 'application/json, text/javascript, */*; q=0.01',
 'HTTP_ACCEPT_ENCODING': 'gzip, deflate',
 'HTTP_ACCEPT_LANGUAGE': 'en-GB,en;q=0.5',
 'HTTP_CACHE_CONTROL': 'no-cache',
 'HTTP_CONNECTION': 'keep-alive',
 'HTTP_CONTENT_LENGTH': '92',
 'HTTP_CONTENT_TYPE': 'application/x-www-form-urlencoded; charset=UTF-8',
 'HTTP_COOKIE': 'phpbb3_a9mfv_u=2759; phpbb3_a9mfv_k=31e5b68a4d3f5896; phpbb3_a9mfv_sid=e804e7bdcfddebcbb684fb6e98fccb1a; style_cookie=null; csrftoken=CFDjgGaVwzFkmVwBxRDBt0OMRe1ClEx4; _ga=GA1.2.483423307.1416590819; sessionid_merge=4borjmxvx1kpwfnpveua37thh3zi211r',
 'HTTP_HOST': 'trout.o...y.org',
 'HTTP_PRAGMA': 'no-cache',
 'HTTP_REFERER': 'https://trout.o...y.org/merge/webclient/',
 'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.8; rv:33.0) Gecko/20100101 Firefox/33.0',
 'HTTP_X_CSRFTOKEN': 'CFDjgGaVwzFkmVwBxRDBt0OMRe1ClEx4',
 'HTTP_X_REQUESTED_WITH': 'XMLHttpRequest',
 'PATH_INFO': u'/webclient/edit_channel_names/253141/',
 'QUERY_STRING': '',
 'REQUEST_METHOD': 'POST',
 'SCRIPT_INFO': '/merge',
 'SCRIPT_NAME': u'/merge',
 'SERVER_NAME': 'ome-c6100-3',
 'SERVER_PORT': '443',
 'SERVER_PROTOCOL': 'HTTP/1.1',
 'wsgi.errors': <flup.server.fcgi_base.TeeOutputStream object at 0x6a4d290>,
 'wsgi.input': <flup.server.fcgi_base.InputStream object at 0x6868090>,
 'wsgi.multiprocess': True,
 'wsgi.multithread': False,
 'wsgi.run_once': False,
 'wsgi.url_scheme': 'https',
 'wsgi.version': (1, 0)}>
```

**Comment by pwalczysko** [comment]
I cannot repeat the bug https://github.com/openmicroscopy/openmicroscopy/pull/3733#issuecomment-95104427 -  all the rest of the UI workflows are fine.

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OME-5.1-merge-push#942](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/942/). See the [console output](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/942/consoleText) for more details.
Possible conflicts:
  - PR #3732 will-moore 'Right panel MissingPyramidException'
    - components/tools/OmeroPy/test/integration/gatewaytest/test_missing_pyramid.py

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OME-5.1-merge-push#943](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/943/). See the [console output](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/943/consoleText) for more details.
Possible conflicts:
  - PR #3732 will-moore 'Right panel MissingPyramidException'
    - components/tools/OmeroPy/test/integration/gatewaytest/test_missing_pyramid.py

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OME-5.1-merge-push#944](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/944/). See the [console output](https://ci.openmicroscopy.org/job/OME-5.1-merge-push/944/consoleText) for more details.
Possible conflicts:
  - PR #3732 will-moore 'Right panel MissingPyramidException'
    - components/tools/OmeroPy/test/integration/gatewaytest/test_missing_pyramid.py

**Comment by will-moore** [comment]
@sbesson That last commit should fix failure in https://ci.openmicroscopy.org/view/Failing/job/OMERO-5.1-merge-integration-python/428/testReport/test.integration.gatewaytest.test_image/TestImage/testProperties/

**Comment by pwalczysko** [comment]
Works fine from UI perspective. The bug https://github.com/openmicroscopy/openmicroscopy/pull/3733#issuecomment-95104427 is not there anymore.

**Comment by mtbc** [comment]
The test looks to be fixed. :+1:

**Comment by sbesson** [comment]
Thanks all. Merging.

**Pull request merged**